azure/services/roleassignments/roleassignments.go

```diff
@@ -160,9 +161,9 @@ func (s *Service) getVMSSPrincipalID(ctx context.Context) (*string, error) {
 	if err != nil {
 		return nil, errors.Wrap(err, "failed to get principal ID for VMSS")
 	}
-	resultVMSS, ok := resultVMSSIface.(compute.VirtualMachineScaleSet)
+	resultVMSS, ok := resultVMSSIface.(armcompute.VirtualMachineScaleSet)
```

Comment:
Are tests failing on main without this change now that scalesets are using SDK v2?

Reply:
They haven't, no, which is...interesting. Let's see if they pass here.